[BUG] udf_test failed in databricks 11.3 ENV

**Describe the bug**
related to https://github.com/rapidsai/cudf/issues/12764
we made some workaround to specify python 3.8 to get pytest working in db113
but failed pyspark pandas version check
```
 AssertionError: incorrect pandas version during required testing partially initialized module 'pandas' has no attribute '__version__' (most likely due to a circular import
```

```
[2023-02-13T11:51:24.238Z] ERROR ../../src/main/python/udf_cudf_test.py - AssertionError: incorrect pandas version during required testing partially ...

[2023-02-13T11:51:24.239Z] ERROR ../../src/main/python/udf_test.py - AssertionError: incorrect pandas version during required testing partially ...

[2023-02-13T11:51:24.239Z] ERROR ../../src/main/python/udf_cudf_test.py - AssertionError: incorrect pandas version during required testing partially ...

[2023-02-13T11:51:24.239Z] ERROR ../../src/main/python/udf_test.py - AssertionError: incorrect pandas version during required testing partially ...

[2023-02-13T11:51:24.239Z] ERROR ../../src/main/python/udf_cudf_test.py - AssertionError: incorrect pandas version during required testing partially ...

[2023-02-13T11:51:24.239Z] ERROR ../../src/main/python/udf_cudf_test.py - AssertionError: incorrect pandas version during required testing partially ...

[2023-02-13T11:51:24.239Z] ERROR ../../src/main/python/udf_test.py - AssertionError: incorrect pandas version during required testing partially ...

[2023-02-13T11:51:24.239Z] ERROR ../../src/main/python/udf_test.py - AssertionError: incorrect pandas version during required testing partially ...
```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
Pass the test

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/jenkins/databricks/init_cudf_udf.sh
+++ b/jenkins/databricks/init_cudf_udf.sh
@@ -18,7 +18,7 @@
 # The initscript to set up environment for the cudf_udf tests on Databricks
 # Will be automatically pushed into the dbfs:/databricks/init_scripts once it is updated.
 
-set -ex
+set -x
 
 CUDF_VER=${CUDF_VER:-23.04}
 CUDA_VER=${CUDA_VER:-11.0}
@@ -26,9 +26,7 @@ CUDA_VER=${CUDA_VER:-11.0}
 # Need to explicitly add conda into PATH environment, to activate conda environment.
 export PATH=/databricks/conda/bin:$PATH
 # Set Python for the running instance
-# PYTHON_VERSION=$(${PYSPARK_PYTHON} -c 'import sys; print("{}.{}".format(sys.version_info.major, sys.version_info.minor))')
-# TODO: revert this change after https://github.com/rapidsai/cudf/issues/12764 is resolved
-PYTHON_VERSION='3.8'
+PYTHON_VERSION=$(${PYSPARK_PYTHON} -c 'import sys; print("{}.{}".format(sys.version_info.major, sys.version_info.minor))')
 
 base=$(conda info --base)
 # Create and activate 'cudf-udf' conda env for cudf-udf tests
@@ -38,8 +36,7 @@ conda create -y -n cudf-udf -c conda-forge python=$PYTHON_VERSION mamba && \
 
 # Use mamba to install cudf-udf packages to speed up conda resolve time
 conda install -y -c conda-forge mamba python=$PYTHON_VERSION
-# Do not error out "This operation will remove conda without replacing it with another version of conda." for now
-${base}/envs/cudf-udf/bin/mamba remove -y c-ares zstd libprotobuf pandas || true
+${base}/envs/cudf-udf/bin/mamba remove -y c-ares zstd libprotobuf pandas
 
 REQUIRED_PACKAGES=(
   cudatoolkit=$CUDA_VER

